uw-frame-components/portal/widgets/controllers.js

```diff
@@ -400,6 +400,98 @@ define(['angular'], function(angular) {
     initializeActionItems();
   }])
 
+  // VARIABLE CONTENT widget type
+  .controller('VariableContentController', [
+    '$scope', '$filter', '$log', function($scope, $filter, $log) {
+    /**
+     * Display provided custom template
+     *   - provisional: Check if it's the annual benefits enrollment
+     *   widget and display markup specific to that widget instead
+     * @param template The provided custom angular/HTML template
+     * @param start The day/time this content became active
+     * @param end The date/time this content will go away
+     */
+    var displayVariableContent = function(template, start, end) {
+      $scope.startDate = start;
+      $scope.endDate = end;
+      if ($scope.widget.fname === 'university-staff-benefits-statement') {
+        configureAnnualBenefitsEnrollmentContent();
+      } else {
+        $scope.template = template;
+      }
+    };
+
+    /**
+     *
+     */
+    var configureAnnualBenefitsEnrollmentContent = function() {
```

Comment:
To generalize this so it doesn't need annual-benefits-enrollment-specific stuff: have it compute the days until the next template switch and expose that integer to the template.

As in, it's currently showing the before-benefit-enrollment-starts content. And it knows the date on which it'll switch to the next template. So it computes the quantity days until that happens, and exposes it to the current template as `daysRemainingShowingThisTemplate` or `daysUntilNextTemplate` or whatever.  And then in the actual ABE content, the before-enrollment template uses, references this variable to get at the count.

So then the ABE start date arrives. Boing, onto next template. And that template gets the count of days until the template after it. So the during-ABE content could include the countdown until the end of ABE.

Turtles all the way down.

Reply:
Unfortunately, the before-benefit-enrollment-starts is still custom content that is specific to ABE, for now. As it's built in its simplest form, this type shows either a basic widget (large icon) or a custom template (based on a date range), so even though we know when the template will switch, we can't communicate about it until the template *does* switch. 

For ABE (and possible for other use cases down the road), we need two ranges of of dates: One for when the template switches on and off, and one for when the "active" period within the custom template starts and ends. 

I'm in favor of rethinking this so that it doesn't just accept any random custom template. Like you said before, there are probably other cases where widgets want to tell someone that they have a task to complete in [x] days, or that it's ongoing, or than it recently ended. That being the case, we could make this a more rigid type that communicates about a few different states and has a single call to action.

Reply:
One thing that doesn't account for is the warning that the active period is *going* to begin. ABE widget has the countdown during the active period, but also has a "Begins [given date]" case leading up to it.